### DISCUSSION TITLE
Variable flow Schedules?

### DISCUSSION THREAD

**Discussion by JCoder01** [tagyoureit/nodejs-poolController#798] [Q&A] [answered]
I have an Intellipro VSF that I just set up. I have the pump direct connected to the controller, with Nixie single body. I like to run the pump at different flow rates at different parts of the day. I was able to do this with schedules on the pump, but I can't seem to figure out how to do with njsPC. I also use home assistant if that would make it easier.
Thanks!

**Comment by rstrouse** [comment, marked as answer]
Add a feature for each speed then assign these to the pump with the speed you want.  The simply create a schedule for the features you want to run.

As a rule of thumb assign the pool circuit to the slowest speed needed to run the clhlorinator.  If you have solar then set the speed required for that.  If you have a cleaner then choose that speed.  And whatever speeds needed to run any of the features.

For the schedules create a pool schedule that runs over the entire regular cycle.  Overlay all other equipment schedules over that.

**Reply by JCoder01** [reply]
Great! Thank you for the quick reply. 
Unrelated follow up. Prior to setting this up I already have a temp sensor and my heater set up via an esp32. Is there anyway to get njsPC/REM talking to them via WiFi or do I need to swap out the ESP for an SBC?

**Reply by rstrouse** [reply]
Most folks run njsPC, dashPanel, and REM on a single SBC out on the pad.  This puts the hardware control local and provides many more options for hardware by stacking hats.  While a bit of socket code could get you there on esp for the sensor and heater relay, much of the other equipment folks use in REM would overshoot the limits of the current esp chips.

Here is the Nixie hardware wiki.  Bear in mind it outlines some of the more common hardware folks are using but it isn't the only way to skin the cat.
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller

**Reply by JCoder01** [reply]
I'm hesitant to switch due to rather limited needs, I just have 1 pump, 1 heater, and 1 water temp sensor. I was having some luck with using the various API's and python to read the data from the ESP and post it to njsPC, but it doesn't seem to get me all the way. I can get `waterSensor1` to update but that doesn't seem to impact the body temperature. Any ideas there? If it's not really possible I guess I'll be ordering a check SBC.

**Reply by rstrouse** [reply]
The pool temp will only update when the pool circuit is engaged.  So if you are updating waterSensor1 and the pool has a green button next to it the pool temp should update.  So when the temp is grayed out it is the last recorded temp for the pool since this is the closest you will get to the actual temperature in the water.  

The real issue is trying to run a heater without all the relay protections and thermostat switching that is in njsPC.  It coordinates everything including the pump speeds required to run the heater.
